src/main/java/org/terasology/launcher/packages/PackageManager.java

```diff
@@ -94,44 +97,62 @@ public void sync() {
         localRepository.saveCache();
     }
 
-    // TODO: Move to constructor
-    public void initDatabase(Path launcherDir, Path gameDir) {
+    public void setupDirs(Path launcherDir, Path gameDir) {
         cacheDir = launcherDir.resolve(CACHE_DIRECTORY);
         installDir = gameDir.resolve(INSTALL_DIRECTORY);
-        final Path sourcesFile = launcherDir.resolve(SOURCES_FILENAME);
+        sourcesFile = launcherDir.resolve(SOURCES_FILENAME);
+    }
 
-        // Copy default sources file if necessary
+    public void validateSources() {
         if (Files.notExists(sourcesFile)) {
-            logger.info("Sources file not found. Copying default file to {}", sourcesFile);
-            try {
-                Files.copy(getClass().getResourceAsStream(SOURCES_FILENAME), sourcesFile);
-            } catch (IOException e) {
-                logger.error("Failed to copy default sources file to {}", sourcesFile);
-            }
-        } else {
-            validateSchema(sourcesFile);
+            logger.warn("sources.json not found: {}", sourcesFile);
+            throw new PackageManagerException("sources.json is missing");
         }
+        if (!validateSchema(sourcesFile)) {
+            throw new PackageManagerException("There were errors reading sources.json");
```

Comment:
This should be localized as well instead of hard-coded English.
Also, this should mention what happens if confirmation is denied (could also be done in the `ContentText` af the `Alert`).